### DISCUSSION TITLE
Clarification on Model Update after Outlier Reduction with Images-Only

### DISCUSSION THREAD

**Discussion by owwdesilva** [MaartenGr/BERTopic#2055] [Q&A] [answered]
Hi @MaartenGr, Hi All,

Clarification on an issue encountered with the  model using an images-only training approach. Below is the relevant code snippet we used to train the model:

```
hdbscan_model = HDBSCAN(min_cluster_size=best_hyperparameters['min_cluster_size'], metric='euclidean', prediction_data=True)
vectorizer = CountVectorizer(stop_words="english", ngram_range=best_hyperparameters['n_gram_range'])

topic_model = BERTopic(
        embedding_model=MultiModalBackend, 
        representation_model=representation_model,
        umap_model=umap_model,
        hdbscan_model=hdbscan_model,
        vectorizer_model=vectorizer,
        ctfidf_model=ctfidf_model,
        calculate_probabilities= True,
        verbose=True,
        top_n_words=best_hyperparameters['top_n_words']
)

topics, probs = topic_model.fit_transform(documents=None, images=images, embeddings=image_embeddings)
```

We observed outliers in the output and proceeded with the outlier reduction method as follows:

`new_topics = topic_model.reduce_outliers(None, topics, images=images, probabilities=probs, strategy="probabilities")
`

However, we encountered an issue when attempting to update the model, as the `update_topics` method requires documents to be passed:

`topic_model.update_topics([], images=images, topics=new_topics)`

Given that our model is trained using only images, providing documents in this context is not feasible. Could you please provide advice or guidance on how to address this issue?

**Comment by MaartenGr** [comment, marked as answer]
This is interesting as I haven't used this images-only procedure with `.reduce_outliers` and `.update_topics`. In all honesty, based on the code I'm seeing in `.update_topics`, I don't think this will work unfortunately. Instead, you could use [manual topic modeling](https://maartengr.github.io/BERTopic/getting_started/manual/manual.html) using the topics you have previously created.

**Reply by owwdesilva** [reply]
Thanks @MaartenGr for the insight. I'll try manual topic modeling with the existing topics.
